x/compliance/types/message_certify_model_test.go

```diff
@@ -1,42 +1,148 @@
+				CertificationDate:     testconstants.CertificationDate.Format(time.RFC3339),
+				CertificationType:     testconstants.CertificationType,
+			},
+			err: validator.ErrFieldLowerBoundViolated,
+		},
+		{
+			name: "vid < 0",
+			msg: MsgCertifyModel{
+				Signer:                sample.AccAddress(),
+				Vid:                   -1,
+				Pid:                   1,
+				SoftwareVersionString: testconstants.SoftwareVersionString,
+				CertificationDate:     testconstants.CertificationDate.Format(time.RFC3339),
+				CertificationType:     testconstants.CertificationType,
+			},
+			err: validator.ErrFieldLowerBoundViolated,
+		},
+		{
+			name: "vid > 65535",
+			msg: MsgCertifyModel{
+				Signer:                sample.AccAddress(),
+				Vid:                   65536,
+				Pid:                   1,
+				SoftwareVersionString: testconstants.SoftwareVersionString,
+				CertificationDate:     testconstants.CertificationDate.Format(time.RFC3339),
+				CertificationType:     testconstants.CertificationType,
+			},
+			err: validator.ErrFieldUpperBoundViolated,
+		},
+		{
+			name: "pid is 0",
+			msg: MsgCertifyModel{
+				Signer:                sample.AccAddress(),
+				Pid:                   0,
+				Vid:                   1,
+				SoftwareVersionString: testconstants.SoftwareVersionString,
+				CertificationDate:     testconstants.CertificationDate.Format(time.RFC3339),
+				CertificationType:     testconstants.CertificationType,
+			},
+			err: validator.ErrFieldLowerBoundViolated,
+		},
+		{
+			name: "pid < 0",
+			msg: MsgCertifyModel{
+				Signer:                sample.AccAddress(),
+				Pid:                   -1,
+				Vid:                   1,
+				SoftwareVersionString: testconstants.SoftwareVersionString,
+				CertificationDate:     testconstants.CertificationDate.Format(time.RFC3339),
+				CertificationType:     testconstants.CertificationType,
+			},
+			err: validator.ErrFieldLowerBoundViolated,
+		},
+		{
+			name: "pid > 65535",
+			msg: MsgCertifyModel{
+				Signer:                sample.AccAddress(),
+				Pid:                   65536,
+				Vid:                   1,
+				SoftwareVersionString: testconstants.SoftwareVersionString,
+				CertificationDate:     testconstants.CertificationDate.Format(time.RFC3339),
+				CertificationType:     testconstants.CertificationType,
+			},
+			err: validator.ErrFieldUpperBoundViolated,
+		},
+		{
+			name: "cd version number  > 65535",
+			msg: MsgCertifyModel{
+				Signer:                sample.AccAddress(),
+				CDVersionNumber:       65536,
+				Pid:                   1,
+				Vid:                   1,
+				SoftwareVersionString: testconstants.SoftwareVersionString,
+				CertificationDate:     testconstants.CertificationDate.Format(time.RFC3339),
+				CertificationType:     testconstants.CertificationType,
+			},
+			err: validator.ErrFieldUpperBoundViolated,
+		},
+	}
+
+	positive_tests := []struct {
+		name string
+		msg  MsgCertifyModel
+	}{
+		{
 			name: "valid address",
```

Comment:
Please find a better description, as not only address, but all fields are valid here.
Also please set  `SoftwareVersion` explicitly.